test/parallel/test-cluster-fork-windowsHide.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const child_process = require('child_process');
+
+let patchedFork = child_process.fork;
+child_process.fork = (...args) => patchedFork.apply(this, args);
+const cluster = require('cluster');
```

Comment:
You can get rid of the monkey-patch now because ultimately it doesn't how matter how the option is passed down, as long as the observable effect is the same.

Presumably the `child_process.spawn()` call can go too?  I admit I don't quite understand why you run the tests in a child process.

Reply:
I keep the monkey-patch just in case some one run the tests on other platforms or in a windows service session. Not sure if this is necessary or practically useful.  

It seems Windows only allocate a new console window for an attaching process spawned by a detached process.  

If process D was spawned by process C with `detached: true` ,  then process W spawned by process D with `detached: false` will get a new black console window popped up.

if D was spawned by C with `detached: false` or W spawned by D with `detached: true`, than no console window will pop up for W.

C = Command,  D = Daemon,  W = worker, practically.

That why the test need to be run in a (detached) child process.

Reply:
@bnoordhuis Is this OK?

Reply:
> I keep the monkey-patch just in case some one run the tests on other platforms or in a windows service session. Not sure if this is necessary or practically useful.

Neither, I'd say. :-)

Can you add a comment to the test that explains why it spawns a child process first?  I get it now from your explanation but it's not obvious from just looking at the code.

Reply:
Sure, i'll add the comment, and remove the monkey-patch thing.